src/Tasks/Microsoft.NET.Build.Tasks/PrepareForReadyToRunCompilation.cs

```diff
@@ -112,20 +115,33 @@ protected override void ExecuteCore()
                 CrossgenTool.SetMetadata("DiaSymReader", _diasymreaderPath);
 
             // Process input lists of files
-            ProcessInputFileList(FilesToPublish, _compileList, _symbolsCompileList, _r2rFiles);
+            ProcessInputFileList(Assemblies, _compileList, _symbolsCompileList, _r2rFiles, _r2rReferences);
         }
 
-        void ProcessInputFileList(ITaskItem[] inputFiles, List<ITaskItem> imageCompilationList, List<ITaskItem> symbolsCompilationList, List<ITaskItem> r2rFilesPublishList)
+        void ProcessInputFileList(ITaskItem[] inputFiles, List<ITaskItem> imageCompilationList, List<ITaskItem> symbolsCompilationList, List<ITaskItem> r2rFilesPublishList, List<ITaskItem> r2rReferenceList)
         {
             if (inputFiles == null)
             {
                 return;
             }
+            
+            var exclusionSet = ExcludeList == null ? null : new HashSet<string>(ExcludeList, StringComparer.OrdinalIgnoreCase);
 
             foreach (var file in inputFiles)
             {
-                if (!InputFileEligibleForCompilation(file))
+                var eligibility = GetInputFileEligibility(file, exclusionSet);
+
+                if (eligibility == Eligibility.None)
+                {
+                    continue;
+                }
+
+                r2rReferenceList.Add(file);
```

Comment:
I don't think this is correct because the `GetInputFileEligitility` now returns `Eligibility.ReferenceOnly` for reference assemblies, and we can't reference those. Crossgen needs to reference implementation assemblies.

Reply:
I'm thinking of the case where we get those references assemblies published under the refs folder. Just wanting to make sure they don't show up here

Reply:
The refs folder would not be here because this is no longer just everything in in resolved files to publish. The set of assemblies are supposed to be known implementation assemblies before we get this far. But I decided to be careful and still check that they are actually assemblies. So it makes sense to be careful there too.